src/util/promise_utils.js

```diff
@@ -62,7 +64,8 @@ function iterate(array, func) {
         return P.try(() => func(array[i], i, array)).then(next);
     }
 
-    return P.try(next).return(results);
+    return P.try(next)
+        .then(() => results);
```

Comment:
```suggestion
    return Promise.resolve().then(next).then(() => results);
```

Reply:
we don't want bluebird here..

Reply:
sure, in that function we still have 2 blubirds:
https://github.com/noobaa/noobaa-core/blob/9b6ec6bf3263d2cb531ec2237f73a0128c9bf184/src/util/promise_utils.js#L39
and 
https://github.com/noobaa/noobaa-core/blob/9b6ec6bf3263d2cb531ec2237f73a0128c9bf184/src/util/promise_utils.js#L62

should I leave them?